lib/Semantics/expression.cpp

```diff
@@ -238,8 +238,10 @@ MaybeExpr ExpressionAnalyzer::CompleteSubscripts(ArrayRef &&ref) {
   if (subscripts == 0) {
     // nothing to check
   } else if (subscripts != symbolRank) {
-    Say("Reference to rank-%d object '%s' has %d subscripts"_err_en_US,
-        symbolRank, symbol.name(), subscripts);
+    if (symbolRank != 0) {
+      Say("Reference to rank-%d object '%s' has %d subscripts"_err_en_US,
```

Comment:
It would be good to have a test case that produces this message, although it looks like there's never been a test case for this condition.

Reply:
Good idea. There was "Reference to rank-0 object..." error that I removed, so I've added a couple of tests.